pkg/crc/preflight/preflight_ubuntu.go

```diff
@@ -0,0 +1,75 @@
+// +build linux
+
+package preflight
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	crcos "github.com/code-ready/crc/pkg/os"
+)
+
+var ubuntuPreflightChecks = []Check{
+	{
+		configKeySuffix:    "check-apparmor-profile-setup",
+		checkDescription:   "Checking if AppArmor is configured",
+		check:              checkAppArmorConfigured(ioutil.ReadFile, constants.MachineCacheDir),
+		fixDescription:     "Updating AppArmor configuration",
+		fix:                fixAppArmorConfigured(ioutil.ReadFile, crcos.WriteToFileAsRoot, constants.MachineCacheDir),
+		cleanupDescription: "Cleaning up AppArmor configuration",
+		cleanup:            restoreAppArmorConfiguration(ioutil.ReadFile, crcos.WriteToFileAsRoot, constants.MachineCacheDir),
+	},
+}
+
+const (
+	appArmorTemplate = "/etc/apparmor.d/libvirt/TEMPLATE.qemu"
+	appArmorHeader   = "profile LIBVIRT_TEMPLATE flags=(attach_disconnected) {"
+)
+
+type reader func(filename string) ([]byte, error)
+type writer func(reason, content, filepath string, mode os.FileMode) error
+
+func checkAppArmorConfigured(reader reader, cacheDir string) func() error {
+	return func() error {
+		template, err := reader(appArmorTemplate)
+		if err != nil {
+			return err
+		}
+		if !strings.Contains(string(template), expectedLines(cacheDir)) {
+			return errors.New("AppArmor profile not configured")
+		}
+		return nil
+	}
+}
+
+func fixAppArmorConfigured(reader reader, writer writer, cacheDir string) func() error {
+	return updateAppArmorTemplate(reader, writer, appArmorHeader, expectedLines(cacheDir))
+}
+
+func restoreAppArmorConfiguration(reader reader, writer writer, cacheDir string) func() error {
+	return updateAppArmorTemplate(reader, writer, expectedLines(cacheDir), appArmorHeader)
+}
+
```

Comment:
This is basically inserting `$dir/*/crc.qcow2` in `/etc/apparmor.d/libvirt/TEMPLATE.qemu` after the `profile LIBVIRT_TEMPLATE flags=(attach_disconnected) {` header (or removing this same line in cleanup). Not sure how this can be made easier to understand, but as this is, it's  far from obvious what this code is doing. Ideally the API would be `insertLine` and `removeLine`, not `update(left, right)`
I think you can hardcode `constants.MachineCacheDir`, the unit test only cares about the string value?